lib/matplotlib/markers.py

```diff
@@ -175,11 +175,12 @@ def __init__(self, marker=None, fillstyle=None):
         fillstyle : string, optional, default: 'full'
             'full', 'left", 'right', 'bottom', 'top', 'none'
         """
-        # The fillstyle has to be set here as it might be accessed by calls to
-        # _recache() in set_marker.
-        self._fillstyle = fillstyle
-        self.set_marker(marker)
+        # The _recache() is called in both set_fillstyle() and set_marker()
+        # methods that's why here we have to have either marker or fillstyle
+        # preinitialized before setting other.
+        self._marker = None
```

Comment:
It is probably better from a readability stand point to initialize all of the state here, rather than relying on a call to `_recache` to set it all up.